guppylang/prelude/builtins.py

```diff
@@ -505,77 +504,35 @@ def __trunc__(self: float) -> float: ...
 
 @guppy.extend_type(list_type_def)
 class List:
-    @guppy.hugr_op(unsupported_op("Append"))
-    def __add__(self: list[T], other: list[T]) -> list[T]: ...
-
-    @guppy.hugr_op(unsupported_op("IsEmpty"))
-    def __bool__(self: list[T]) -> bool: ...
-
-    @guppy.hugr_op(unsupported_op("Contains"))
-    def __contains__(self: list[T], el: T) -> bool: ...
-
-    @guppy.hugr_op(unsupported_op("AssertEmpty"))
-    def __end__(self: list[T]) -> None: ...
-
-    @guppy.hugr_op(unsupported_op("Lookup"))
-    def __getitem__(self: list[T], idx: int) -> T: ...
+    @guppy.hugr_op(unsupported_op("pop"))  # TODO: unwrap and swap None
+    def __getitem__(self: list[L], idx: int) -> L: ...
 
-    @guppy.hugr_op(unsupported_op("IsNotEmpty"))
-    def __hasnext__(self: list[T]) -> tuple[bool, list[T]]: ...
-
-    @guppy.custom(NoopCompiler())
-    def __iter__(self: list[T]) -> list[T]: ...
-
-    @guppy.hugr_op(unsupported_op("Length"))
-    def __len__(self: list[T]) -> int: ...
+    @guppy.hugr_op(unsupported_op("pop"))  # TODO: check None and unwrap
+    def __setitem__(self: list[L], idx: int, value: L @ owned) -> None: ...
 
-    @guppy.hugr_op(unsupported_op("Repeat"))
-    def __mul__(self: list[T], other: int) -> list[T]: ...
-
-    @guppy.hugr_op(unsupported_op("Pop"))
-    def __next__(self: list[T]) -> tuple[T, list[T]]: ...
+    @guppy.hugr_op(unsupported_op("length"))  # TODO: inout return in wrong order
```

Comment:
is this ordering being resolved elsewhere?

Reply:
Actually turns out that the length op doesn't return the list at all! https://github.com/CQCL/hugr/issues/1543